update: Add criteria to VPN Services page and update other sections (#2788)
privacyguides/privacyguides.org: docs/vpn.md

| Provider | Countries | WireGuard | Port Forwarding | IPv6 | Anonymous Payments
|---|---|---|---|---|---

> | [Proton](#proton-vpn) | 112+ | :material-check:{ .pg-green } | :material-information-outline:{ .pg-blue } Partial Support | :material-information-outline:{ .pg-blue } Partial Support | Cash

Reviewer: ```suggestion
| [Proton](#proton-vpn) | 112+ | :material-check:{ .pg-green } | :material-alert-outline:{ .pg-orange } Partial Support | :material-information-outline:{ .pg-blue } Limited Support | Cash
```

> #### :material-alert-outline:{ .pg-orange } No IPv6 Support

Reviewer 2: ```suggestion
#### :material-alert-outline:{ .pg-orange } Limited IPv6 Support
```

> Proton [now supports IPv6](https://protonvpn.com/support/prevent-ipv6-vpn-leaks/) in their browser extension but only 80% of their servers are IPv6-compatible. On the other platforms, the Proton VPN applications will block all outgoing IPv6 traffic, so you don't have to worry about your IPv6 address being leaked, but you will not be able to connect to any IPv6-only sites, and you will not be able to connect to Proton VPN from an IPv6-only network.

Reviewer 2: ```suggestion
Proton [now supports IPv6](https://protonvpn.com/support/prevent-ipv6-vpn-leaks/) in their browser extension and Linux client. On the other platforms, the Proton VPN applications will block all outgoing IPv6 traffic, so you don't have to worry about your IPv6 address being leaked, but you will not be able to connect to any IPv6-only sites, and you will not be able to connect to Proton VPN from an IPv6-only network.
```

Reviewer 2: I think stating the exact percentage of servers that offer IPv6 support isn't as important as saying which platforms support it. Plus it'll get outdated quickly anyway.

Author: About the Linux client supporting ipv6, I purposely didn't mention it, because in my testing it didn't work (they had the ipv6 kill-switch enabled despite ipv6 being enabled in their app...)

#### :material-information-outline:{ .pg-info } Remote Port Forwarding

Proton VPN currently only supports ephemeral remote [port forwarding](https://protonvpn.com/support/port-forwarding) via NAT-PMP, with 60 second lease times. The Windows app provides an easy to access option for it, while on other operating systems you'll need to run your own [NAT-PMP client](https://protonvpn.com/support/port-forwarding-manual-setup). Torrent applications often support NAT-PMP natively.

#### :material-information-outline:{ .pg-blue } Anti-Censorship

> Always have a backup method, in case [censors break your current technique](https://github.com/net4people/bbs/issues/384#issue-2461515121).

Reviewer 2: ```suggestion
```

Not very helpful since it doesn't specify what backup method to use. I'd rather just keep this focused on Proton's feature.

Author: This is important so people have for example two VPNs. But maybe this could be clearer.

Reviewer 2: I think just leave it out, maybe it could go somewhere else.

Reviewer 3: > https://github.com/net4people/bbs/issues/384#issue-2461515121

Has anyone posted about this on our forum? It'd be nice if we talked about this topic there, where people would be able to seek assistance or ask questions.

Author: github.com/net4people/bbs is already a forum about censorship circumvention. Unfortunately, our audience isn´t very large in highly-censored countries AFAIK. There was a[ topic about censorship evasion](https://discuss.privacyguides.net/t/do-mullvad-and-ivpn-have-any-censorship-resistant-technologies/21462/17), but it didn´t get much traction.

I think it is better to link to BBS, as they are more up to date with censorship.

> I think just leave it out, maybe it could go somewhere else. It's also pretty repetitive having the same thing three times.

Could it be put in the introduction?

Reviewer 4: Yeah I think it would fit better there.

Reviewer 3: I wonder if it would fit better as informational content on [`basics/vpn-overview.md`](https://github.com/privacyguides/privacyguides.org/blob/main/docs/basics/vpn-overview.md) instead of this page.

Author: I was thinking anti-censorship might deserve a separate page in Basics. Thoughts?

Reviewer 4: Maybe. This should go on a page like that I think rather than here.

IVPN previously supported port forwarding, but removed the option in [June 2023](https://ivpn.net/blog/gradual-removal-of-port-forwarding). Missing this feature could negatively impact certain applications, especially peer-to-peer applications like torrent clients.

#### :material-check:{ .pg-green } Anti-Censorship

> Always have a backup method, in case [censors break your current technique](https://github.com/net4people/bbs/issues/384#issue-2461515121).

Reviewer 2: ```suggestion
```

Same as above

> IVPN has obfuscation modes using the [v2ray](https://v2ray.com/en/index.html) protocol which helps in situations where VPN protocols like OpenVPN or Wireguard are blocked. Currently this feature is only available on Desktop and [iOS](https://ivpn.net/knowledgebase/ios/v2ray). It has two modes where it can use [VMess](https://guide.v2fly.org/en_US/basics/vmess.html) over QUIC or TCP connections. QUIC is a modern protocol with better congestion control and therefore may be faster with reduced latency. The TCP mode makes your data appear as regular HTTP traffic.

Reviewer 2: ```suggestion
IVPN has obfuscation modes using the [v2ray](https://v2ray.com/en/index.html) project which helps in situations where VPN protocols like OpenVPN or Wireguard are blocked. Currently this feature is only available on Desktop and [iOS](https://ivpn.net/knowledgebase/ios/v2ray). It has two modes where it can use [VMess](https://guide.v2fly.org/en_US/basics/vmess.html) over QUIC or TCP connections. QUIC is a modern protocol with better congestion control and therefore may be faster with reduced latency. The TCP mode makes your data appear as regular HTTP traffic.
```

I can't find anywhere in the v2ray documentation where they refer to it as a protocol.

Author: ![2630](https://github.com/user-attachments/assets/5659413a-46b5-4440-b1b1-1953d6d6f90a)
Saying that you can use a project doesn't really make sense IMO. But I will not fight over it.

Reviewer 2: How about just saying "using v2ray"?

Author: Make sense

![Mullvad logo](assets/img/vpn/mullvad.svg){ align=right }

> **Mullvad** is a fast and inexpensive VPN with a serious focus on transparency and security. They have been in operation since 2009. Mullvad is based in Sweden and offers a 30-day money-back guarantee for payments methods that allow it.

Reviewer 2: ```suggestion
**Mullvad** is a fast and inexpensive VPN with a serious focus on transparency and security. They have been in operation since 2009. Mullvad is based in Sweden and offers a 30-day money-back guarantee for payment methods that allow it.
```

#### :material-check:{ .pg-green } Accepts Cash and Monero

> Mullvad, in addition to accepting credit/debit cards and PayPal, accepts Bitcoin, Bitcoin Cash, **Monero** and **cash/local currency** as anonymous forms of payment. Prepaid cards with redeem codes are also available. Mullvad also accepts Swish and bank wire transfers, as well as a few european payment systems.

Reviewer 2: ```suggestion
Mullvad, in addition to accepting credit/debit cards and PayPal, accepts Bitcoin, Bitcoin Cash, **Monero** and **cash/local currency** as anonymous forms of payment. Prepaid cards with redeem codes are also available. Mullvad also accepts Swish and bank wire transfers, as well as a few European payment systems.
```

Mullvad previously supported port forwarding, but removed the option in [May 2023](https://mullvad.net/en/blog/2023/5/29/removing-the-support-for-forwarded-ports). Missing this feature could negatively impact certain applications, especially peer-to-peer applications like torrent clients.

#### :material-check:{ .pg-green } Anti-Censorship

> Always have a backup method, in case [censors break your current technique](https://github.com/net4people/bbs/issues/384#issue-2461515121).

Reviewer 2: ```suggestion
```

Same as above.

Mullvad offers several features to help bypass censorship and access the internet freely:

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Reviewer: Also see https://github.com/privacyguides/privacyguides.org/pull/2788#discussion_r1820000693 for the updated section that includes a short description of DAITA

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Author: > I couldn't find much info about "UDP-over-TCP" in the Mullvad docs, but I think it's the same as Shadowsocks + OpenVPN?

No, it's a different option, and is only available for Wireguard. It was the only option on mobile before they introduced Wireguard (WG) over Shadowsocks.

>Other anti-censorship techniques by Proton VPN and IVPN? 
Yes, mainly toward Proton as IVPN has some V2RAY.
>Is the content in the linked GitHub issue still relevant?
Mostly, but China actually doesn't block Shadowsocks, although this doesn't mean it couldn't happen.

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Reviewer: > No, it's a different option, and is only available for Wireguard. It was the only option on mobile before they introduced Wireguard (WG) over Shadowsocks.

Ahh, now I think I see it. The "Obfuscation" mode is UDP-over-TCP, while "Bridge mode" is OpenVPN through Shadowsocks. Thanks for the clarification.

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Reviewer: @I-I-IT I updated my suggestion based on your feedback.

I decided to leave out the bit about Mullvad's obfuscation methods being more effective than Proton's, since there's already a section in the Proton anti-censorship section which disclaims that Stealth isn't as effective as circumventing sophisticated filtering. I also included the paper you linked as a replacement for the GitHub issue.

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Author: >* **Obfuscation modes**: Mullvad has two built-in obfuscation modes: UDP-over-TCP (the "Obfuscation" option), OpenVPN via Shadowsocks (the "Bridge mode" option only available on their desktop apps), 

Let me take screenshots so you see better how those modes are.
On mobile:
![2647](https://github.com/user-attachments/assets/2c08280d-19bd-4b5b-8703-7a85883373d4)

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Reviewer: Thanks for all the helpful screenshots! I just reverted this section back to the original + updated and added the last line. I'll defer to you for what more info to include and how to format it. Feel free to change it as you see fit.

> * **Obfuscation modes**: Mullvad has two built-in obfuscation modes: "UDP-over-TCP" and ["Wireguard over Shadowsocks"](https://mullvad.net/en/blog/introducing-shadowsocks-obfuscation-for-wireguard). These modes disguise your VPN traffic as regular internet traffic, making it harder for censors to detect and block.

Author: Actually, how can I edit your suggestion while keeping you as a coauthor?

* **Custom server IPs**: To counter IP-blocking, you can request custom server IPs from Mullvad's support team. Once you receive the custom IPs, you can input the text file in the "Server IP override" settings, which will override the chosen server IP addresses with ones that aren't known to the censor.
* **Bridges and proxies**: Mullvad also allows you to use bridges or proxies to reach their API (needed for authentication), which can help bypass censorship attempts that block access to the API itself.

#### :material-check:{ .pg-green } Mobile Clients

Mullvad has published [App Store](https://apps.apple.com/app/id1488466513) and [Google Play](https://play.google.com/store/apps/details?id=net.mullvad.mullvadvpn) clients, both supporting an easy-to-use interface as opposed to requiring you to manually configure your WireGuard connection. The Android client is also available on [GitHub](https://github.com/mullvad/mullvadvpn-app/releases).

#### :material-information-outline:{ .pg-blue } Additional Notes

> Mullvad is very transparent about which nodes they [own or rent](https://mullvad.net/en/servers). They use [ShadowSocks](https://shadowsocks.org) in their ShadowSocks + OpenVPN (Desktop) and Wireguard over Shadowsocks configuration, making them more resistant against firewalls with [Deep Packet Inspection](https://en.wikipedia.org/wiki/Deep_packet_inspection) trying to block VPNs. Supposedly, [China has to use a different method to block ShadowSocks servers](https://github.com/net4people/bbs/issues/22).

Author: I agree, it is a bit of a duplicate. Do I approved this change and then modify it, or would you rather modify you suggestion?

> Mullvad is very transparent about which nodes they [own or rent](https://mullvad.net/en/servers). They use [ShadowSocks](https://shadowsocks.org) in their ShadowSocks + OpenVPN (Desktop) and Wireguard over Shadowsocks configuration, making them more resistant against firewalls with [Deep Packet Inspection](https://en.wikipedia.org/wiki/Deep_packet_inspection) trying to block VPNs. Supposedly, [China has to use a different method to block ShadowSocks servers](https://github.com/net4people/bbs/issues/22).

Reviewer: Yeah, I can modify this suggestion, and then you can go over it to see if the information is correct (since you have more knowledge than I do about Shadowsocks and DAITA, given that you've talked about both on the forum).

> Mullvad is very transparent about which nodes they [own or rent](https://mullvad.net/en/servers). They use [ShadowSocks](https://shadowsocks.org) in their ShadowSocks + OpenVPN (Desktop) and Wireguard over Shadowsocks configuration, making them more resistant against firewalls with [Deep Packet Inspection](https://en.wikipedia.org/wiki/Deep_packet_inspection) trying to block VPNs. Supposedly, [China has to use a different method to block ShadowSocks servers](https://github.com/net4people/bbs/issues/22).

Author: Seems good to me.

- Killswitch built in to clients.
- Multihop support. Multihopping is important to keep data private in case of a single node compromise.
- If VPN clients are provided, they should be [open source](https://en.wikipedia.org/wiki/Open_source), like the VPN software they generally have built into them. We believe that [source code](https://en.wikipedia.org/wiki/Source_code) availability provides greater transparency about what your device is actually doing.

> - Basic censorship resistance that can bypass firewalls without DPI.

Reviewer 4: ```suggestion
- Censorship resistance features
```
I don't think we can prove whether these features can bypass any given firewall with or without DPI so I think we should just broaden it a bit to just include anti-censorship features.

**Best Case:**

- Killswitch with highly configurable options (enable/disable on certain networks, on boot, etc.)
- Easy-to-use VPN clients

> - Supports [IPv6](https://en.wikipedia.org/wiki/IPv6). We expect that servers will allow incoming connections via IPv6 and allow you to access services hosted on IPv6 addresses.

Reviewer: ```suggestion
- [IPv6](https://en.wikipedia.org/wiki/IPv6) support. We expect that servers will allow incoming connections via IPv6 and allow you to access services hosted on IPv6 addresses.
```

- Capability of [remote port forwarding](https://en.wikipedia.org/wiki/Port_forwarding#Remote_port_forwarding) assists in creating connections when using P2P ([Peer-to-Peer](https://en.wikipedia.org/wiki/Peer-to-peer)) file sharing software or hosting a server (e.g., Mumble).

> - Obfuscation technology which camouflages the true nature of internet traffic, allowing users to circumvent advanced internet censorship through methods like DPI.

Reviewer: ```suggestion
- Obfuscation technology which camouflages the true nature of internet traffic, allowing users to circumvent advanced internet censorship methods like DPI.
```

> - Obfuscation technology which camouflages the true nature of internet traffic, allowing users to circumvent advanced internet censorship methods like DPI.

Reviewer 4: ```suggestion
- Obfuscation technology which camouflages the true nature of internet traffic.
```
Same here, I don't know how we can prove it can bypass any given firewall or DPI implementation.

- Killswitch built in to clients.
- Multihop support. Multihopping is important to keep data private in case of a single node compromise.
- If VPN clients are provided, they should be [open source](https://en.wikipedia.org/wiki/Open_source), like the VPN software they generally have built into them. We believe that [source code](https://en.wikipedia.org/wiki/Source_code) availability provides greater transparency about what your device is actually doing.

> -Basic censorship resistance that can bypass firewalls without DPI.

Reviewer 2: ```suggestion
- Basic censorship resistance that can bypass firewalls without DPI.
```

**Best Case:**

- Killswitch with highly configurable options (enable/disable on certain networks, on boot, etc.)
- Easy-to-use VPN clients
- Supports [IPv6](https://en.wikipedia.org/wiki/IPv6). We expect that servers will allow incoming connections via IPv6 and allow you to access services hosted on IPv6 addresses.
- Capability of [remote port forwarding](https://en.wikipedia.org/wiki/Port_forwarding#Remote_port_forwarding) assists in creating connections when using P2P ([Peer-to-Peer](https://en.wikipedia.org/wiki/Peer-to-peer)) file sharing software or hosting a server (e.g., Mumble).

> - Obfuscation technology which camouflage the true nature of the internet traffic, allowing user to circumvent advanced internet censorship with DPI.

Author: This thought, is not completely correct. (Sorry to be nitpicky). 
"Regular internet traffic" is ambiguous, and can go from HTTP(S) traffic to video calling. Shadowsocks specifically disguise VPN traffic as HTTP traffic.
Also, I would rather say users than customers.

> - Obfuscation technology which camouflage the true nature of the internet traffic, allowing user to circumvent advanced internet censorship with DPI.

Reviewer: No need to apologize. In that case, my suggestion for this part is just for typos.

- Strong Encryption Schemes: OpenVPN with SHA-256 authentication; RSA-2048 or better handshake; AES-256-GCM or AES-256-CBC data encryption.
- Forward Secrecy.
- Published security audits from a reputable third-party firm.

> - VPN servers have full-disk encryption or are RAM-only.

Reviewer: ```suggestion
- VPN servers that use full-disk encryption or are RAM-only.
```

**Best Case:**

- Strongest Encryption: RSA-4096.
- Optional quantum-resistant encryption.
- Forward Secrecy.
- Comprehensive published security audits from a reputable third-party firm.
- Bug-bounty programs and/or a coordinated vulnerability-disclosure process.

> - VPN servers are RAM-only.

Reviewer: ```suggestion
- RAM-only VPN servers.
```

**Minimum to Qualify:**

- Public-facing leadership or ownership.

> - The company is based in a jurisdiction where it cannot be forced to do secret logging.

Reviewer: ```suggestion
- Company based in a jurisdiction where it cannot be forced to do secret logging.
```

### Additional Functionality

> While not strictly requirements, there are some factors we looked into when determining which providers to recommend. These include content blocking functionality, warrant canaries, excellent customer support, the number of allowed simultaneous connections, etc.

Reviewer 2: ```suggestion
While not strictly requirements, there are some factors we looked into when determining which providers to recommend. These include content blocking functionality, warrant canaries, multi-hop connections, excellent customer support, the number of allowed simultaneous connections, etc.
```

Reviewer 2: Not sure why this was removed?

Author: Because it is already a criteria, so it is a duplicate to put it here.
